arraycontext/container/traversal.py

```diff
@@ -27,6 +27,8 @@
 ~~~~~~~~~~~~~~~~
 .. autofunction:: from_numpy
 .. autofunction:: to_numpy
+.. autofunction:: flatten_to_numpy
+.. autofunction:: unflatten_from_numpy
```

Comment:
`flatten_contaier_to_numpy` and `unflatten_numpy_to_container` actually could work too.

Reply:
Those sound good to me! I'll wait for some more feedback, but otherwise we can rename them like that.

Reply:
Added a sentence to the docs to mention that this is required in 17f74ad.

Let me know that's sufficient / phrased unambiguously enough.

Reply:
Agreed.

One issue is that they do slightly different things:
* these flatten the whole container into one array
* the ones in `meshmode.dof_array` traverse the container and only flatten the `DOFArray`s.

Not sure if anyone depends on that functionality though? So it could be safe to deprecate and point to these functions instead.

Reply:
Thanks for spotting that! I don't think anyone is relying on the "just the `DOFArray`s" aspect of those. (I'm not even sure in what scenario that would be useful.) Plus, if desired, it can be replicated with relative ease atop these new functions.

Reply:
Yeah, that sounds good! I'll add some deprecation warnings in there after this is in.